src/pd/cluster.rs

```diff
@@ -204,22 +211,23 @@ impl Connection {
 
         warn!("updating pd client, blocking the tokio core");
         let start = Instant::now();
-        let (client, members) = self.try_connect_leader(&old_cluster.members, timeout)?;
-        let tso = TimestampOracle::new(old_cluster.id, &client)?;
-
+        let (client, members) = self.try_connect_leader(previous_members, timeout).await?;
+        let tso = TimestampOracle::new(cluster_id, &client)?;
+        let last_update = Instant::now();
         let cluster = Cluster {
-            id: old_cluster.id,
+            id: cluster_id,
+            last_update,
             client,
             members,
             tso,
         };
-        *self.last_update.write().unwrap() = Instant::now();
+        *self.last_update.write().unwrap() = last_update;
```

Comment:
Can we just remove the `last_update` in `Connection`? It looks always identical to the one in `Cluster`.